inline-java build failure

Pinging @alpmestan. Also, the URL in the .cabal file gave me a 404 on Github.

```
> /tmp/stackage-build8$ stack unpack inline-java-0.1
Unpacked inline-java-0.1 to /tmp/stackage-build8/inline-java-0.1/
> /tmp/stackage-build8/inline-java-0.1$ runghc -clear-package-db -global-package-db -package-db=/home/stackage/work/builds/nightly/pkgdb Setup configure --package-db=clear --package
ackage-db=/home/stackage/work/builds/nightly/pkgdb --libdir=/home/stackage/work/builds/nightly/lib --bindir=/home/stackage/work/builds/nightly/bin --datadir=/home/stackage/work/buil
re --libexecdir=/home/stackage/work/builds/nightly/libexec --sysconfdir=/home/stackage/work/builds/nightly/etc --docdir=/home/stackage/work/builds/nightly/doc/inline-java-0.1 --html
kage/work/builds/nightly/doc/inline-java-0.1 --haddockdir=/home/stackage/work/builds/nightly/doc/inline-java-0.1 --flags= --extra-lib-dirs /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/
-extra-include-dirs /usr/lib/jvm/java-8-openjdk-amd64/include
Configuring inline-java-0.1...
> /tmp/stackage-build8/inline-java-0.1$ runghc -clear-package-db -global-package-db -package-db=/home/stackage/work/builds/nightly/pkgdb Setup build
Building inline-java-0.1...
Preprocessing library inline-java-0.1...
[1 of 3] Compiling Foreign.JNI.Types ( src/Foreign/JNI/Types.hs, dist/build/Foreign/JNI/Types.o )

src/Foreign/JNI/Types.hs:57:56: error:
    Module ‘Data.Singletons’ does not export ‘KProxy(..)’
```

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -2979,6 +2979,7 @@ expected-test-failures:
     - hedis
     - hocilib # oracle
     - hworker
+    - jvm
     - katip-elasticsearch # elasticsearch
     - log # ElasticSearch
     - mangopay # https://github.com/prowdsponsor/mangopay/issues/30

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -1565,6 +1565,9 @@ packages:
         - distributed-closure
         - H
         - inline-r
+        - jni
+        - jvm
+        - sparkle
         - th-lift
 
     "Christopher Reichert <creichert07@gmail.com> @creichert":
@@ -2319,8 +2322,6 @@ packages:
     "Alp Mestanogullari <alpmestan@gmail.com> @alpmestan":
         - taggy
         - taggy-lens
-        # 0.1 Compilation failure https://github.com/fpco/stackage/issues/1660 - inline-java
-        # via inline-java - sparkle
 
     "Alex McLean <lxyxpx@gmail.com> @yaxu":
         - tidal
@@ -2807,11 +2808,15 @@ configure-args:
   - /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server
   - --extra-include-dirs
   - /usr/lib/jvm/java-8-openjdk-amd64/include
+  - --extra-include-dirs
+  - /usr/lib/jvm/java-8-openjdk-amd64/include/linux
   sparkle:
   - --extra-lib-dirs
   - /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server
   - --extra-include-dirs
   - /usr/lib/jvm/java-8-openjdk-amd64/include
+  - --extra-include-dirs
+  - /usr/lib/jvm/java-8-openjdk-amd64/include/linux
   hocilib:
   - --extra-lib-dirs
   - /usr/local/lib

